code/game/machinery/scan_gate.dm

```diff
@@ -51,16 +61,26 @@
 
 /obj/machinery/scanner_gate/Crossed(atom/movable/AM)
 	. = ..()
-	auto_scan(AM)
+	if(!blocking)
+		auto_scan(AM)
+
+/obj/machinery/scanner_gate/Bumped(atom/movable/AM)
+	. = ..()
+	if(blocking)
+		auto_scan(AM)
 
 /obj/machinery/scanner_gate/proc/auto_scan(atom/movable/AM)
 	if(!(stat & (BROKEN|NOPOWER)) && isliving(AM))
 		perform_scan(AM)
 
-/obj/machinery/scanner_gate/proc/set_scanline(type, duration)
+/obj/machinery/scanner_gate/proc/set_scanline(stype, duration)
```

Comment:
why rename the var name if it's still gonna be confusing

Reply:
`type` refers to a datum's path, so I changed it to `stype`, but to make it more digestible I will rename it to `scan_type`